alibi/explainers/partial_dependence.py

```diff
@@ -242,13 +242,21 @@ def explain(self,  # type: ignore[override]
                 )
             )
 
+        # extract the number of targets that the PD/ICE was computed for
+        key = Kind.AVERAGE if kind in [Kind.AVERAGE, Kind.BOTH] else Kind.INDIVIDUAL
+        n_targets = pds[0][key].shape[0]
+
         if self.target_names is None:
             # set the `target_names` when the user did not provide the target names
             # we do it here to avoid checking model's type, prediction function etc.
-            key = Kind.AVERAGE if kind in [Kind.AVERAGE, Kind.BOTH] else Kind.INDIVIDUAL
-            n_targets = pds[0][key].shape[0]
             self.target_names = [f'c_{i}' for i in range(n_targets)]
 
+        elif len(self.target_names) != n_targets:
+            logger.warning('The length of `target_names` does not match the number of predicted outputs. '
+                           'Ensure that the lengths match, otherwise a call to the `plot_pd` method might '
+                           'raise an error or produce undesired labeling.')
```

Comment:
I don't think there is a possibility to check whether the number of targets that we computed the PD/ICE for matches the length of `target_names` without a dummy call. Altough the user should be aware of that I decided to include a warning message in case that happens. Shall we go even further and raise an error?